src/detect-app-layer-protocol.c

```diff
@@ -78,24 +78,38 @@ static int DetectAppLayerProtocolPacketMatch(
     switch (data->mode) {
         case DETECT_ALPROTO_DIRECTION:
             if (p->flowflags & FLOW_PKT_TOSERVER) {
+                if (f->alproto_ts == ALPROTO_UNKNOWN)
```

Comment:
Should be done as in #11723 in only one place

Reply:
I don't think that works, it's using all kinds of different versions of `alproto_*` here

Reply:
You are right